tempesta_fw/sock.c

```diff
@@ -194,6 +195,9 @@ ss_do_send(struct sock *sk, SsSkbList *skb_list)
 	       smp_processor_id(), __func__,
 	       sk, tcp_send_head(sk), sk->sk_state);
 
+	if (flags & SS_SEND_F_CONN_CLOSE)
+		return;
```

Comment:
What's going on here? It seems if we're sending data with connection close flag, then the function just frees the skb list (ever w/o freeing skbs!) and just exits.

Also please remove comment line with MSG_MORE at 180th line. I agree that we don't need MSG_MORE flag since we do just one TCP call for whole message.